actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -144,17 +144,21 @@ def initialize(parameters = {})
     end
 
     # Returns true if another +Parameters+ object contains the same content and
-    # permitted flag, or other Hash-like object contains the same content. This
-    # override is in place so you can perform a comparison with `Hash`.
-    def ==(other_hash)
-      if other_hash.respond_to?(:permitted?)
-        super
+    # permitted flag.
+    def ==(other)
+      if other.respond_to?(:permitted?)
+        self.permitted? == other.permitted? && self.parameters == other.parameters
```

Comment:
Ah, now I understand. Considering a test case:
```rb
  test 'is equal to an unpermitted Parameters instance with same params' do
    params1 = ActionController::Parameters.new(a: 1).permit(:a)
    params2 = ActionController::Parameters.new(a: 1)
    assert(params1 == params2)
    assert(params2 == params1)
  end
```
it would fail if we just compared on `to_h`, but pass if comparing only `parameters`. Actually, all tests will pass if I just remove the `permitted?` check. Is that what we should do, then?